cloud/status.go

```diff
@@ -44,3 +84,34 @@ func (s *StatusService) GetAllStatuses(ctx context.Context) ([]Status, *Response
 
 	return statusList, resp, nil
 }
+
+func (s *StatusService) SearchStatusesPaginated(ctx context.Context, tweaks ...searchF) ([]JiraStatus, *Response, error) {
+	apiEndpoint := "rest/api/3/statuses/search"
+
+	search := []searchParam{}
+	for _, f := range tweaks {
+		search = f(search)
+	}
+
+	queryString := ""
+	for _, param := range search {
+		queryString += fmt.Sprintf("%s=%s&", param.name, param.value)
+	}
+
+	if queryString != "" {
+		apiEndpoint += "?" + queryString
```

Comment:
Would it make more sense to use use url.Values{}? You could add each query param individually with v.Set(param.name, param.value) and v.Encode() to get the query string. Right now any values with question marks or ampersands might be an issue.